app-office/texmacs/texmacs-1.99.12.recipe

```diff
@@ -0,0 +1,111 @@
+SUMMARY="A scientific editing platform designed to create beautiful technical documents"
+DESCRIPTION="TeXmacs provides a unified and user friendly framework for editing \
+structured documents with different types of content: text, mathematics, graphics, \
+interactive content, slides, etc.\
+Its rendering engine uses high-quality typesetting algorithms so as to produce \
+professionally looking documents, which can either be printed out or presented from \
+a laptop."
+HOMEPAGE="http://www.texmacs.org"
+COPYRIGHT="Free Software Foundation"
+LICENSE="GNU GPL v3"
+
+SOURCE_URI="https://github.com/mgubi/texmacs/archive/texmacs-$portVersion-haiku-qt5.tar.gz"
+CHECKSUM_SHA256="607fa4db4676b78780f7af086feec2e2332dc18a371d79fa50d640de6db10b43"
+SOURCE_DIR="texmacs-texmacs-$portVersion-haiku-qt5/src"
+PATCHES="texmacs-$portVersion.patchset"
+
+REVISION="1"
+
+ADDITIONAL_FILES="texmacs.rdef.in"
+DISABLE_SOURCE_PACKAGE="yes"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	texmacs$secondaryArchSuffix = $portVersion
+	app:TeXmacs
+	cmd:fig2ps$secondaryArchSuffix
+	"
+
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libQt5Concurrent$secondaryArchSuffix
+	lib:libQt5Core$secondaryArchSuffix
+	lib:libQt5Gui$secondaryArchSuffix
+	lib:libQt5Svg$secondaryArchSuffix
+	lib:libQt5Widgets$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
+	lib:libgmp$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
+	lib:libltdl$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libiconv$secondaryArchSuffix
+	devel:libQt5Concurrent$secondaryArchSuffix
+	devel:libQt5Core$secondaryArchSuffix
+	devel:libQt5Gui$secondaryArchSuffix
+	devel:libQt5Svg$secondaryArchSuffix
+	devel:libQt5Widgets$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	devel:libguile$secondaryArchSuffix
+	devel:libfreetype$secondaryArchSuffix
+	devel:libgmp$secondaryArchSuffix
+	devel:libltdl$secondaryArchSuffix
+	"
+
+BUILD_PREREQUIRES="
+	cmd:bc
+	cmd:autoconf
+	cmd:automake
+	cmd:find
+	cmd:gcc$secondaryArchSuffix
+	cmd:gettext$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:python2.7
+	cmd:sed
+	"
+
+BUILD()
+{
+#	./autogen.sh
+	runConfigure ./configure
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+
+	rm $binDir/texmacs
+
+	mkdir $appsDir
+	mkdir $appsDir/lib
+	mv $libDir/TeXmacs/bin/texmacs.bin $appsDir/TeXmacs
+	cp /system/lib/libguile.so* $appsDir/lib/.
```

Comment:
Why would you do that?

Reply:
You should add ```lib:libguile >= versionnumber``` instead to the REQs.

Reply:
I think mine is the right solution to the following problem: Guile 1.8 and Guile 2.2 cannot be installed at the same time (they both provide cmd:guile and lib:libguile ). So I require guile 1.8 only while building and copy the library with the program. So that the end used do not have to have guile 1.8 installed on the system and can use more modern guile while keeping TeXmacs functioning. It works on my machine.

Reply:
Not sure what the libversion for 2.2 is, but in your PR for guile 1.8 there is a libVersionCompat available: https://github.com/haikuports/haikuports/pull/4811/files#diff-c48bf806855e70b29ab17493d383c123R38

Reply:
This point is not clear to me (and maybe my PR for guile1 is not completely correct): I think the libraries are compatible but the point is that there are differences in the interpretation of the Scheme programs which breaks some programs. This is why I need the older version. Since is not a common problem and likely the end user would never use Guile 1.8 nowadays I was thinking that have a runtime dependence on the older version is not the optimal solution. Also both packages guile and guile1 install cmd:guile, so they have anyway conflicts.

Reply:
Looking at the package for guile 2.2.4 it provides guile-2.2.so, so this shouldn't interfere with the 1.8 version?
1.8 version provides:
```
lib:libguile$secondaryArchSuffix
```
2.2.4 provides:
```
lib:libguile_$portVers$secondaryArchSuffix
```

Reply:
If you would try to link against 2.2.4 you'd have to use "lib:libguild_2.2$secondaryArchSuffix" then ...

Reply:
PS, checking local build (32bit) atm for guile 1.8 and will check this one later :)

Reply:
Still, there is the problem of cmd:guile and cmd:guile-config : both guile1 and guile packages provide it and therefore they will conflict, if I understand this correctly. So the two packages cannot be installed at the same time (and haikuporter tries always to put my guile1 package in obsolete..., which I do not really understand why)

Reply:
I don't think you should have them as a requirement (not tested) if you require the right lib:* and devel:* packages (cmd:guild-config is provided in the devel:* package), will check in a bit ...

Reply:
I entered the IRC channel now if you want to discuss the problem.

Reply:
Copyiing the lib into this package is definetely not an option. Please use the correct req declaration.

Reply:
Also not needed ...

Reply:
not anymore, I've splitted the guile1 package by putting cmd:guile in guile1_tools so that there are no conflicts anymore.